### PR TITLE
Draft: Implemented stamping created time

### LINKED ISSUE #49: oci_image should support stamping and creation time.
currently, the creation time of the images is set to a fixed date for reproducibility.  this should be overridable with stamping.

### PR DESCRIPTION
**Pull request by liningpan**
Implemented stamping container created time with `BUILD_TIMESTAMP`

Closes #49

### CODE REVIEW

**Comment by thesayyn** [comment]
this goes against our pattern of provide us a file, which may or may not be stamped and we will used that file instead of teaching every attribute stamping.

**Comment by liningpan** [comment]
So, does it mean you would prefer to have the created time provided in a file?

**Comment by thesayyn** [comment]
yes that's the approach we've taken.

**Comment by liningpan** [comment]
Okay, we can set the created timestamp from a file now. I'm not sure what's the best way to add tests. Finding the file containing the timestamp requires one level of indirection (i.e. need to read the manifest file to find the blob).

Also, because timestamp is a single string, I don't know if there is a way to distinguish it from a label, which makes the semantics of this parameter different from others.

File: examples/created_timestamp/BUILD [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        "BUILD_TIMESTAMP",
+    ],
+)
+
+expand_template(
+    name = "build_timestamp",
+    out = "build_timestamp.txt",
+    stamp_substitutions = {"BUILD_TIMESTAMP": "{{BUILD_TIMESTAMP}}"},
+    substitutions = {"BUILD_TIMESTAMP": "946684800"},
+    template = ":build_timestamp_tmpl",
```
Comment (alexeagle) [inline examples/created_timestamp/BUILD:25]: Nit: newer version of this rule accepts a list of strings here which become the lines of the template, avoiding need for the write_file rule

**Review by alexeagle — COMMENTED** [review]
Feels like the right shape, maybe we would be okay with making the oci_image macro do more of this so there's less BUILD file for users to write

**Comment by thesayyn** [comment]
@liningpan do you mind rebasing this PR against 2.x, and rename the attribute to `created` to follow the spec?

**Comment by thesayyn** [comment]
Closing this as inactive. Feel free to submit a new one

**Pull request closed without merging**